src/test/java/bio/terra/service/filedata/azure/tables/TableFileDaoTest.java

```diff
@@ -1,105 +1,119 @@
-    when(tableServiceClient.getTableClient(any())).thenReturn(tableClient);
-    when(tableClient.getEntity(PARTITION_KEY, FILE_ID)).thenReturn(entity);
-    when(tableClient.getEntity(PARTITION_KEY, "nonexistentFile"))
-        .thenThrow(TableServiceException.class);
+  private TableEntity createTableEntity(boolean hasMimeType) {
+    TableEntity entity =
+        new TableEntity(PARTITION_KEY, FILE_ID)
+            .addProperty(FireStoreFile.FILE_ID_FIELD_NAME, FILE_ID)
+            .addProperty(FireStoreFile.DESCRIPTION_FIELD_NAME, "A test entity")
+            .addProperty(FireStoreFile.BUCKET_RESOURCE_ID_FIELD_NAME, "bucketResourceId")
+            .addProperty(FireStoreFile.LOAD_TAG_FIELD_NAME, "loadTag")
+            .addProperty(FireStoreFile.FILE_CREATED_DATE_FIELD_NAME, "fileCreatedDate")
+            .addProperty(FireStoreFile.GS_PATH_FIELD_NAME, "gsPath")
+            .addProperty(FireStoreFile.CHECKSUM_CRC32C_FIELD_NAME, "checksumCrc32c")
+            .addProperty(FireStoreFile.CHECKSUM_MD5_FIELD_NAME, "checksumMd5")
+            .addProperty(FireStoreFile.SIZE_FIELD_NAME, 1L);
+    if (hasMimeType) {
+      entity.addProperty(FireStoreFile.MIME_TYPE_FIELD_NAME, "application/json");
+    }
+    return entity;
   }
 
   @Test
-  public void testRetrieveFileMetadata() {
-    FireStoreFile fileMetadata = dao.retrieveFileMetadata(tableServiceClient, DATASET_ID, FILE_ID);
+  void testFromTableEntityMissingMimeType() {
+    TableEntity entity = createTableEntity(false);
     FireStoreFile expected = FireStoreFile.fromTableEntity(entity);
-    assertEquals("The same object is returned", fileMetadata, expected);
+    assertNull(expected.getMimeType(), "Entity mimeType does not throw NPE");
   }
 
-  @Test
-  public void testDeleteFileMetadata() {
-    boolean exists = dao.deleteFileMetadata(tableServiceClient, DATASET_ID, FILE_ID);
-    assertTrue("Existing row is deleted", exists);
-    boolean result = dao.deleteFileMetadata(tableServiceClient, DATASET_ID, "nonexistentFile");
-    assertFalse("Non-existent row is not deleted", result);
-  }
+  @Nested
+  class TestRetrieveDeleteFileMetadata {
 
-  @Test
-  public void testBatchRetrieveFileMetadata() {
-    FireStoreDirectoryEntry fsDirectoryEntry = new FireStoreDirectoryEntry().fileId(FILE_ID);
-    List<FireStoreDirectoryEntry> directoryEntries = List.of(fsDirectoryEntry);
-    List<FireStoreFile> expectedFiles = List.of(FireStoreFile.fromTableEntity(entity));
-    List<FireStoreFile> files =
-        dao.batchRetrieveFileMetadata(tableServiceClient, DATASET_ID, directoryEntries);
-    assertEquals(
-        "A file record is found for each directory entry", files.size(), expectedFiles.size());
-    assertEquals("The same object is returned", files.get(0), expectedFiles.get(0));
+    @Mock private TableServiceClient tableServiceClient;
+    @Mock private TableClient tableClient;
+    @Autowired private TableFileDao dao;
```

Comment:
In general you only want to use `@Autowired` in a test that's testing MVC or database operations. `TableFileDao` doesn't use either MVC or database operations so you don't need to autowire it, you can instead create a copy for testing using `new()`. Once you've done that, you can remove these annotations from the outer class:
```
@SpringBootTest
@AutoConfigureMockMvc
@EmbeddedDatabaseTest
```

```suggestion
    private final TableFileDao dao = new TableFileDao(mock(ExecutorService.class));
```